projects/plugins/jetpack/extensions/shared/external-media/sources/jetpack-app-media.js

```diff
@@ -0,0 +1,111 @@
+import { QRCode } from '@automattic/jetpack-components';
+import { useCallback, useEffect, useState } from '@wordpress/element';
+import { __, sprintf, _n } from '@wordpress/i18n';
+import { JetpackAppIcon } from '../../icons';
+import useInterval from '../../use-interval';
+import MediaBrowser from '../media-browser';
+import { MediaSource } from '../media-service/types';
+import withMedia from './with-media';
+
+function JetpackAppMedia( props ) {
+	const { media, insertMedia, isCopying, multiple, getMedia } = props;
+
+	const wpcomBlogId = window?.Jetpack_Editor_Initial_State?.wpcomBlogId || 0;
+	const imagePath = window?.Jetpack_Editor_Initial_State?.pluginBasePath + '/images/';
+
+	// get the current time and store it in the state
+	const [ currentTime ] = useState( Date.now() / 1000 );
+	const getNextPage = useCallback( () => {
+		getMedia( `/wpcom/v2/app-media?refresh=true&after=${ currentTime }`, true );
+	}, [ getMedia, currentTime ] );
+
+	const getNextPagePull = useCallback( () => {
+		getMedia( `/wpcom/v2/app-media?refresh=true&after=${ currentTime }`, false, false );
+	}, [ getMedia, currentTime ] );
+
+	const onCopy = useCallback(
+		items => {
+			insertMedia( items );
+		},
+		[ insertMedia ]
+	);
+	if ( media.length === 1 && ! multiple ) {
```

Comment:
Should this be wrapped in a `useEffect`?

Reply:
Great idea! Thanks